stdlib/public/Platform/Darwin.swift

```diff
@@ -14,3 +14,18 @@
 
 public let MAP_FAILED =
   UnsafeMutableRawPointer(bitPattern: -1)! as UnsafeMutableRawPointer!
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi' or '.pi' to get the value of correct type and avoid casting.")
+public let M_PI = Double.pi
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 2' or '.pi / 2' to get the value of correct type and avoid casting.")
+public let M_PI_2 = Double.pi / 2
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 4' or '.pi / 4' to get the value of correct type and avoid casting.")
+public let M_PI_4 = Double.pi / 4
+
+@available(swift, deprecated: 3.0, message: "Please use '2.squareRoot()'.")
+public let M_SQRT2 = 2.0.squareRoot()
```

Comment:
Digging into it, I originally had a static inline wrapper around C `sqrt`. However, that ran into https://bugs.swift.org/browse/SR-2089, so it had to be replaced with a non-inline stub [some history here: https://github.com/apple/swift/pull/3454]. We should be able to fix that now that 2089 is addressed, which should also let us constant fold these, I think.